src/main/java/com/iota/iri/service/tipselection/WalkValidator.java

```diff
@@ -1,13 +1,23 @@
 package com.iota.iri.service.tipselection;
 
+import com.iota.iri.controllers.TransactionViewModel;
 import com.iota.iri.model.Hash;
+import com.iota.iri.utils.collections.impl.BoundedHashSet;
+import com.iota.iri.utils.collections.interfaces.BoundedSet;
+
+import java.util.LinkedHashSet;
+import java.util.Set;
 
 /**
  * Validates consistency of tails.
  */
 @FunctionalInterface
 public interface WalkValidator {
 
+    int MAX_CACHE_SIZE = 2_000_000;
```

Comment:
I'd suggest to beware of the tyranny of the default magic number; wouldn't this do best as a configuration parameter with a default value?

Reply:
Yup,
I wanted to get the first iteration of the PR so I only did the necessary changes. I agree with you ofc.